### PR TITLE
Replace arc with kokoro and bazel support for continuous integration.

### PR DESCRIPTION
**Pull request by jverkoey**

### CODE REVIEW

File: .kokoro [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  tests_dir_prefix="github/repo/"
+fi
+
+# Fixes a bug in bazel where objc_library targets have a _ prefix.
+find ${tests_dir_prefix}tests/unit -type f -name '*.swift' -exec sed -i '' -E "s/import Motion(.+)/import _Motion\1/" {} + || true
+# CocoaPods requires that public headers of dependent pods be implemented using framework import
+# notation, while bazel requires that dependency headers be imported with quoted notation.
+find ${tests_dir_prefix}src -type f -name '*.h' -exec sed -i '' -E "s/import <MotionInterchange\/MotionInterchange\.h>/import \"MotionInterchange.h\"/" {} + || true
+stashed_dir=$(pwd)
+reset_imports() {
```
Comment (romoore) [inline .kokoro]: Nit: Prefer to define functions at the top of the file (after global configs).

File: .kokoro [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+# Display commands to stderr.
+set -x
+
+if [ ! -d .kokoro-ios-runner ]; then
+  git clone https://github.com/material-foundation/kokoro-ios-runner.git .kokoro-ios-runner
+fi
+
+pushd .kokoro-ios-runner
+git fetch > /dev/null
+TAG=$(git tag -l "v3*" | sort | tail -n1)
```
Comment (romoore) [inline .kokoro]: Nit: Should the "v3" be a variable at the top of the script?  Is this the kokoro runner tag we want to check out?

Comment (romoore) [inline .kokoro]: I'm  worried about the sorting here.  "3.9" will be sorted *before* "3.11", which would be incorrect.

```bash
rsmoore:~ rsmoore$ cat versions.txt 
3.0.1
3.9
3.9.0
3.09.1
3.0
3.10.1
3.1
3.1.11
3.12.2
3.01.2
rsmoore:~ rsmoore$ sort versions.txt 
3.0
3.0.1
3.01.2
3.09.1
3.1
3.1.11
3.10.1
3.12.2
3.9
3.9.0
rsmoore:~ rsmoore$ 
```

File: .kokoro [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+git checkout $TAG > /dev/null
```
Comment (romoore) [inline .kokoro]: Safety/expansion: Please quote `"$TAG"`

File: .kokoro [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+popd
+
+if [ -z "$KOKORO_BUILD_NUMBER" ]; then
+  tests_dir_prefix=""
+else
+  tests_dir_prefix="github/repo/"
+fi
+
+# Fixes a bug in bazel where objc_library targets have a _ prefix.
+find ${tests_dir_prefix}tests/unit -type f -name '*.swift' -exec sed -i '' -E "s/import Motion(.+)/import _Motion\1/" {} + || true
```
Comment (romoore) [inline .kokoro]: Nit: I believe `"${tests_dir_prefix}tests/unit"` should be quoted.

File: .kokoro [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+# CocoaPods requires that public headers of dependent pods be implemented using framework import
+# notation, while bazel requires that dependency headers be imported with quoted notation.
+find ${tests_dir_prefix}src -type f -name '*.h' -exec sed -i '' -E "s/import <MotionInterchange\/MotionInterchange\.h>/import \"MotionInterchange.h\"/" {} + || true
```
Comment (romoore) [inline .kokoro]: Same quotation request `"${tests_dir_prefix}src"`

File: .kokoro [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+stashed_dir=$(pwd)
+reset_imports() {
+  # Undoes our source changes from above.
+  find ${stashed_dir}/${tests_dir_prefix}tests/unit -type f -name '*.swift' -exec sed -i '' -E "s/import _Motion(.+)/import Motion\1/" {} + || true
```
Comment (romoore) [inline .kokoro]: More quotations, please.

File: BUILD [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    ]),
+    deps = [
+      "@motion_interchange_objc//:MotionInterchange"
+    ],
+    enable_modules = 1,
+    includes = ["src"],
+    visibility = ["//visibility:public"],
+)
+
+load("@build_bazel_rules_apple//apple:swift.bzl", "swift_library")
```
Comment (romoore) [inline BUILD]: Nit/Style: Please place all loads at the top of the file.

File: BUILD [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+    deps = [":MotionAnimator"],
+    visibility = ["//visibility:private"],
+)
+
+objc_library(
+    name = "UnitTestsLib",
+    srcs = glob([
+        "tests/unit/*.m",
+    ]),
+    enable_modules = 1,
```
Comment (romoore) [inline BUILD]: Question: Does it make sense to add headers now or should we wait until we have some to add?

File: BUILD [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+load("@build_bazel_rules_apple//apple:ios.bzl", "ios_unit_test")
+
+ios_unit_test(
+    name = "UnitTests",
+    deps = [
+      ":UnitTestsLib",
+      ":UnitTestsSwiftLib"
+    ],
+    minimum_os_version = "8.0",
```
Comment (romoore) [inline BUILD]: Why does this unit test application have a minimum OS version? Does the component have a different value for `minimum_os_version`?

**Review by romoore — CHANGES_REQUESTED** [review]

File: .kokoro (hunk 2 above)
Comment (jverkoey) [reply .kokoro]: Yeah this is a not super ideal at the moment.

https://stackoverflow.com/questions/21394536/how-to-simulate-sort-v-on-mac-osx gives some insight on a better version sort for mac:

```
cat versions.txt | sort -t. -s -k 1,1n -k 2,2n -k 3,3n -k 4,4n
3.0
3.0.1
3.1
3.01.2
3.1.11
3.9
3.9.0
3.09.1
3.10.1
3.12.2
```

But I'll have to test this on the kokoro instance as well.

File: .kokoro (hunk 3 above)
Comment (jverkoey) [reply .kokoro]: Done.

File: .kokoro (hunk 4 above)
Comment (jverkoey) [reply .kokoro]: Done.

File: .kokoro (hunk 5 above)
Comment (jverkoey) [reply .kokoro]: Done.

File: .kokoro (hunk 1 above)
Comment (jverkoey) [reply .kokoro]: Done.

Comment (jverkoey) [reply .kokoro]: Done.

File: .kokoro (hunk 6 above)
Comment (jverkoey) [reply .kokoro]: Done.

File: BUILD (hunk 7 above)
Comment (jverkoey) [reply BUILD]: Done.

File: BUILD (hunk 8 above)
Comment (jverkoey) [reply BUILD]: Can you clarify what you mean by headers?

Comment (romoore) [reply BUILD]: hdrs = glob(["tests/unit/*.h"]),

File: BUILD (hunk 9 above)
Comment (jverkoey) [reply BUILD]: There isn't a way to specify a minimum OS version for objc_library that I know of.

For a list of examples of usage of this property:
https://github.com/bazelbuild/rules_apple/search?utf8=%E2%9C%93&q=minimum_os_version&type=

File: BUILD (hunk 8 above)
Comment (jverkoey) [reply BUILD]: Ah, yes it probably makes sense to add it now.

Comment (jverkoey) [reply BUILD]: Done.

File: BUILD (hunk 9 above)
Comment (romoore) [reply BUILD]: Yeah, that seems to be the case. https://github.com/bazelbuild/rules_apple/blob/master/apple/ios.bzl

Ack! :D

**Comment by jverkoey** [comment]
Needs review.

File: .kokoro [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+  reset_imports() {
+    # Undoes our source changes from above.
+    find "${stashed_dir}/${tests_dir_prefix}tests/unit" -type f -name '*.swift' -exec sed -i '' -E "s/import _Motion(.+)/import Motion\1/" {} + || true
+    find "${stashed_dir}/${tests_dir_prefix}src" -type f -name '*.h' -exec sed -i '' -E "s/import \"MotionInterchange.h\"/import <MotionInterchange\/MotionInterchange.h>/" {} + || true
+  }
+  trap reset_imports EXIT
+}
+
+if [ ! -d .kokoro-ios-runner ]; then
+  git clone https://github.com/material-foundation/kokoro-ios-runner.git .kokoro-ios-runner
```
Comment (romoore) [inline .kokoro:47]: Do you want to exit early if this clone fails?

File: .kokoro [hunk 11]
```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
```
Comment (romoore) [inline .kokoro:1]: Copyright notice?

File: BUILD [hunk 12]
```diff
@@ -0,0 +1,60 @@
+# Description:
```
Comment (romoore) [inline BUILD:15]: Copyright notice?

File: WORKSPACE [hunk 13]
```diff
@@ -0,0 +1,17 @@
+git_repository(
```
Comment (romoore) [inline WORKSPACE:15]: Copyright notice? Does a WORKSPACE file support such a thing?

File: .kokoro (hunk 11 above)
Comment (jverkoey) [reply .kokoro:1]: Done.

File: .kokoro (hunk 10 above)
Comment (jverkoey) [reply .kokoro:47]: Yep, early exiting is handled by `set -e`.

File: BUILD (hunk 12 above)
Comment (jverkoey) [reply BUILD:15]: Done.

File: WORKSPACE (hunk 13 above)
Comment (jverkoey) [reply WORKSPACE:15]: Done.

File: .kokoro (hunk 10 above)
Comment (romoore) [reply .kokoro:47]: Ack.

**Review by romoore — APPROVED** [review]

**Pull request merged**